### PR TITLE
[ADD] Search feature backend

### PR DESCRIPTION
**Pull request by realsdx**

### CODE REVIEW

**Comment by Compro-Prasad** [comment]
Squash commits and force push.

**Comment by ghoshbishakh** [comment]
code looks good

**Comment by realsdx** [comment]
Updated the PR

File: backend/search/views.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    serializer_class = NoticeSearchSerializer
+    permission_classes = (AllowAny,)
+
+    def get_queryset(self):
+        request = self.request
+        queryset = EmptySearchQuerySet()
+
+        if request.GET.get('q', ''):
+            query = request.GET.get('q', '')
+            queryset =SearchQuerySet().models(Notice).filter(content=query).order_by('-date')
```
Comment (Compro-Prasad) [inline backend/search/views.py]: Space missing after `=`.

File: backend/search/views.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    permission_classes = (AllowAny,)
+
+    def get_queryset(self):
+        request = self.request
+        queryset = EmptySearchQuerySet()
+
+        if request.GET.get('q', ''):
+            query = request.GET.get('q', '')
+            queryset =SearchQuerySet().models(Faculty).filter(content=query)
+            return [i.object for i in queryset]
```
Comment (Compro-Prasad) [inline backend/search/views.py]: Add newline at the end of each python file you have created.

File: backend/search/tests.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
```
Comment (Compro-Prasad) [inline backend/search/tests.py]: Remove this file if there are no tests.

File: backend/search/templatetags/synonym_tags.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    return synonyms
+
+@register.filter
+def add_general_synonyms(text_data):
+    try:
+        text_data = text_data.lower()
+        for syn_list in GENERAL_SYNONYMS:
+            if syn_list[0] in text_data:
+                text_data = text_data.replace(syn_list[0], "".join(syn_list),1)
+                # pattern = r'\b'+syn_list[0]+r'\b'
```
Comment (Compro-Prasad) [inline backend/search/templatetags/synonym_tags.py]: Remove these comments.

File: backend/search/templatetags/synonym_tags.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+GENERAL_SYNONYMS = [
+    ['exam ','examination '],
+    ['examination ','exam '],
+    ['sem ','semester '],
+    ['semester ','sem '],
+    ['b.tech ','b. tech ','btech '],
+    ['b. tech ','b.tech ','btech '],
+]
+
+# GENERAL_SYNONYMS = [
```
Comment (Compro-Prasad) [inline backend/search/templatetags/synonym_tags.py]: Remove these comments.

File: backend/search/templatetags/synonym_tags.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import re
+from django import template
+register = template.Library()
+
+DEPT_SYNONYMS = [
+    ['CS', 'CSE'],
```
Comment (Compro-Prasad) [inline backend/search/templatetags/synonym_tags.py]: Add `IT` too.

File: backend/search/models.py [hunk 7]
```diff
@@ -0,0 +1,3 @@
+from django.db import models
+
+# Create your models here.
```
Comment (Compro-Prasad) [inline backend/search/models.py]: Remove this file.

File: backend/search/admin.py [hunk 8]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (Compro-Prasad) [inline backend/search/admin.py]: Remove this file.

**Comment by realsdx** [comment]
updated the pr

**Comment by Compro-Prasad** [comment]
I am travelling now. Will check it when I will be back.

**Comment by realsdx** [comment]
okay

**Comment by divyanshumehta** [comment]
Looks good..merging!

**Pull request merged**